install.sh

```
@@ -0,0 +1,470 @@
+
+build_from_source() {
+    local repo_dir
+    local build_mode="${1:-release}"
+
+    ensure_dep "git" "brew install git (macOS) or apt install git (Linux)"
+
+    # Install Rust if missing
+    if ! check_dep "cargo"; then
+        info "Installing Rust toolchain..."
+        curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y
+        source "${HOME}/.cargo/env"
```

Shellcheck (0.11.0)

[info] Not following: ./.cargo/env was not specified as input (see shellcheck -x).

(SC1091)